src/layers/tile/infinite.rs

```diff
@@ -39,33 +39,81 @@ impl InfiniteTileLayerData {
         let mut chunks = HashMap::<(i32, i32), Chunk>::new();
         parse_tag!(parser, "data", {
             "chunk" => |attrs| {
-                let chunk = Chunk::new(parser, attrs, e.clone(), c.clone(), tilesets)?;
-                chunks.insert((chunk.x, chunk.y), chunk);
+                let chunk = InternalChunk::new(parser, attrs, e.clone(), c.clone(), tilesets)?;
+                let first_pos = chunk.first_tile_pos;
+                for x in first_pos.0..first_pos.0 + chunk.width as i32 {
+                    for y in first_pos.1..first_pos.1 + chunk.height as i32 {
+                        let chunk_pos = tile_to_chunk_pos(x, y);
+                        let relative_pos = (x - chunk_pos.0 * Chunk::WIDTH as i32, y - chunk_pos.1 * Chunk::HEIGHT as i32);
+                        let chunk_index = (relative_pos.0 + relative_pos.1 * Chunk::WIDTH as i32) as usize;
+                        let internal_pos = (x - first_pos.0, y - first_pos.1);
+                        let internal_index = (internal_pos.0 + internal_pos.1 * chunk.width as i32) as usize;
+
+                        chunks.entry(chunk_pos).or_insert_with(Chunk::new).tiles[chunk_index] = chunk.tiles[internal_index];
+                    }
+                }
                 Ok(())
             }
         });
 
         Ok(Self { chunks })
     }
+
+    pub(crate) fn get_tile(&self, x: i32, y: i32) -> Option<&LayerTileData> {
+        let chunk_pos = tile_to_chunk_pos(x, y);
+        self.chunks
+            .get(&chunk_pos)
+            .and_then(|chunk| {
+                let relative_pos = (
+                    x - chunk_pos.0 * Chunk::WIDTH as i32,
+                    y - chunk_pos.1 * Chunk::HEIGHT as i32,
+                );
+                let chunk_index = (relative_pos.0 + relative_pos.1 * Chunk::WIDTH as i32) as usize;
+                chunk.tiles.get(chunk_index).map(Option::as_ref)
+            })
+            .flatten()
+    }
+}
+
+fn tile_to_chunk_pos(x: i32, y: i32) -> (i32, i32) {
+    (
+        floor_div(x, Chunk::WIDTH as i32),
+        floor_div(y, Chunk::HEIGHT as i32),
+    )
 }
 
 #[derive(Debug, PartialEq, Clone)]
 pub struct Chunk {
-    pub x: i32,
-    pub y: i32,
-    pub width: u32,
-    pub height: u32,
     tiles: Vec<Option<LayerTileData>>,
 }
 
 impl Chunk {
+    pub const WIDTH: u32 = 16;
+    pub const HEIGHT: u32 = 16;
+
+    pub(crate) fn new() -> Self {
+        Self {
+            tiles: vec![None; Self::WIDTH as usize * Self::HEIGHT as usize],
```

Comment:
If we use static size, would it be better to use an array instead of a `Vec`, or is that not helpful regarding moving it around in memory as part of its use in a hash map?

Reply:
True; We could wrap it in a `Box` to ensure moves aren't expensive.